R/methods-MbioDataset.R

```diff
@@ -283,4 +283,119 @@ setMethod("getCollection", "MbioDataset", function(object, collectionName = char
     } 
 
     return(abundanceData)
+})
+
+collectionVarNamesGeneric <- getGeneric("getCollectionVariableNames", "veupathUtils")
+#' Get Microbiome Dataset Collection Variable Names
+#' 
+#' Get the variable names in a collection in the Microbiome Dataset.
+#' 
+#' @examples
+#' variableNames <- getCollectionVariableNames(microbiomeData::DiabImmune, "16S (V4) Genus")
+#' @param object A Microbiome Dataset
+#' @param collectionName The name of the collection to return the variable names for
+#' @return a character vector of the variable names in the requested collection
+#' @export
+#' @rdname getCollectionVariableNames
+#' @aliases getCollectionVariableNames,MbioDataset,character-method
+setMethod(collectionVarNamesGeneric, "MbioDataset", function(object, collectionName) {
+    return(veupathUtils::getCollectionVariableNames(getCollection(object, collectionName)))
+})
+
+#' Get Microbiome Dataset Variables
+#' 
+#' Get the variables in the Microbiome Dataset by their names.
+#' The requested variables could belong to any collection or
+#' to the metadata. The returned data.table will contain the 
+#' requested variables as columns. If one of the requested
+#' variables cannot be returned, a warning will be printed.
+#' 
+#' @examples
+#' getCollectionVariableNames(microbiomeData::DiabImmune, "16S (V4) Genus")
+#' getMetadataVariableNames(microbiomeData::DiabImmune)
+#' variablesDT <- getVariables(
+#'      microbiomeData::DiabImmune, 
+#'      list("metadata" = c("age_months", "sex"),
+#'           "16S (V4) Genus" = "Bacteroides", 
+#'           "WGS Metagenome enzyme pathway abundance data" = "ANAGLYCOLYSIS-PWY: glycolysis III (from glucose)"
+#'      )
+#' )
+#' @param object A Microbiome Dataset
+#' @param variables The names of the variables to return. this should be a named list
```

Comment:
Nit: Capitalize "This" at start of sentence

Reply:
never!